src/_pytest/nodes.py

```diff
@@ -264,7 +276,11 @@ def getparent(self, cls):
         """ get the next parent node (including ourself)
         which is an instance of the given class"""
         current = self
-        while current and not isinstance(current, cls):
```

Comment:
I may have confused myself, but these are not equivalent, I think?

Reply:
its equivalent

Reply:
Are you sure? Let `self: Node`, `cls=NotNode` and `self.parent = None`.

Before it would return `None`. After it would return `self`.

Unless I'm still confused.

Reply:
Correct, will only add the type comment there like you suggested above.